integration/app_integration_test.go

```diff
@@ -336,6 +336,108 @@ func TestAppAccessNoHeaderOverrides(t *testing.T) {
 	}
 }
 
+// TestAppAccessRewriteHeadersRoot validates that http headers from application
+// rewrite configuration are correctly passed to proxied applications in root.
+func TestAppAccessRewriteHeadersRoot(t *testing.T) {
```

Comment:
add a case of rewriting an existing header, like `Host`

Reply:
Done, plus a bunch of other additional cases.